core/state/snapshot/snapshot.go

```diff
@@ -666,6 +666,9 @@ func diffToDisk(bottom *diffLayer) *diskLayer {
 
 // Release releases resources
 func (t *Tree) Release() {
+	t.lock.Lock()
```

Comment:
Please change it to `RLock`, as we only iterate the layer map

Reply:
I guess same for `generating` and `DiskRoot`

Reply:
yep, thanks!